File: Python/project/weather.py
import requests
import argparse

# API URL and key
API_URL = 'https://api.openweathermap.org/data/2.5/weather'
API_KEY = '2bc09c898973d33c3b585075829ee4c0'

# Create a command-line argument parser
parser = argparse.ArgumentParser(description='Get current weather conditions of a city.')
parser.add_argument('city', metavar='C', type=str, help='Name of the city')
args = parser.parse_args()

# Send a GET request to the API with the city name and API key
response = requests.get(API_URL, params={'q': args.city, 'appid': API_KEY})

# If the response is successful (status code 200), print the weather conditions
if response.status_code == 200:
    data = response.json()
    print(f"Current weather in {args.city}:")
    print(f"Description: {data['weather'][0]['description']}")
    print(f"Temperature: {data['main']['temp']} K")
    print(f"Humidity: {data['main']['humidity']}%")
else:
    print(f"Error getting weather data: {response.status_code}")

File: Nodejs project/server/weather.js
const express = require('express');
const cors = require('cors');
const axios = require('axios');

const app = express();
app.use(cors());

const API_KEY = '2bc09c898973d33c3b585075829ee4c0';

app.get('/weather/:lat/:lon', async (req, res) => {
  const { lat, lon } = req.params;
  const url = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${API_KEY}`;
  try {
    const response = await axios.get(url);
    const data = response.data;
    res.json(data);
  } catch (error) {
    console.log(error);
  }
});

app.listen(4000, () => console.log('Server running on port 4000'));
